RecoParticleFlow/PFClusterProducer/interface/PFEcalEndcapRecHitCreator.h

```diff
@@ -0,0 +1,229 @@
+          if(schi<0) EEDetId myID(vec[ic]);
+      
+          theTTDetIds_[tthashedindex]=towid1;
+
+          // check if this SC is already in the list of the corresponding TT
+          std::vector<int>::const_iterator itcheck=find(SCofTT_[tthashedindex].begin(),
+                                                        SCofTT_[tthashedindex].end(),
+                                                        schi);
+          if(itcheck==SCofTT_[tthashedindex].end())
+            SCofTT_[tthashedindex].push_back(schi);
+      
+          // check if this crystal is already in the list of crystals per sc
+          itcheck=find(CrystalsinSC_[schi].begin(),CrystalsinSC_[schi].end(),cellhashedindex);
+          if(itcheck==CrystalsinSC_[schi].end())
+            CrystalsinSC_[schi].push_back(cellhashedindex);
+
+          // check if the TT is already in the list of sc
+          itcheck=find(TTofSC_[schi].begin(),TTofSC_[schi].end(),tthashedindex);
+          if(itcheck==TTofSC_[schi].end())
+            TTofSC_[schi].push_back(tthashedindex);
+        }
+
+    }
+
+ protected:
+
+
+    bool isHighInterest(const EEDetId& detid) {
+
+      int schi=SChashedIndex(detid);
+      // check if it has already been treated or not
+      // 0 <=> not treated
+      // 1 <=> high interest
+      // -1 <=> low interest
+      if(SCHighInterest_[schi]!=0) return (SCHighInterest_[schi]>0);
+
+      // now look if a TT contributing is of high interest
+      const std::vector<int> & tts(TTofSC_[schi]);
+      unsigned size=tts.size();
+      bool result=false;
+      for(unsigned itt=0;itt<size&&!result;++itt)
+        {
+          EcalTrigTowerDetId towid = theTTDetIds_[tts[itt]];
+          if(srFullReadOut(towid)) result=true;
+        }
+      SCHighInterest_[schi]=(result)? 1:-1;
+      theFiredSC_.push_back(schi);
+      return result;
+    }
+
+    bool srFullReadOut(EcalTrigTowerDetId& towid) {
+      EBSrFlagCollection::const_iterator srf = srFlagHandle_->find(towid);
+      if(srf == srFlagHandle_->end()) return false;
+      return ((srf->value() & ~EcalSrFlag::SRF_FORCED_MASK) == EcalSrFlag::SRF_FULL);  
+    }
+
+    edm::EDGetTokenT<EcalRecHitCollection> recHitToken_;
+    edm::EDGetTokenT<EBSrFlagCollection> srFlagToken_;
+
+    const EcalTrigTowerConstituentsMap* eTTmap_;  
+
+    // arraws for the selective readout emulation
+    // fast EEDetId -> TT hashedindex conversion
+    std::vector<int>  towerOf_;
+    // vector of the original DetId if needed
+    std::vector<EcalTrigTowerDetId> theTTDetIds_;
+    // list of SC "contained" in a TT.
+    std::vector<std::vector<int> > SCofTT_;
+    // list of TT of a given sc
+    std::vector<std::vector<int> > TTofSC_;
+    // status of each SC 
+    std::vector<int> SCHighInterest_;
+    // list of fired SC
+    std::vector<int> theFiredSC_;
+    // the list of fired TT
+    std::vector<int> theFiredTTs_;
+    // the cells in each SC
+    std::vector<std::vector<int> > CrystalsinSC_;
```

Comment:
It looks like sizes of this and other vectors (the outer part, at least) are known and fixed.
Please use std::array in this case instead